src/languages/go.js

```diff
@@ -30,7 +30,7 @@ function(hljs) {
         className: 'string',
         variants: [
           hljs.QUOTE_STRING_MODE,
-          {begin: '\'', end: '[^\\\\]\''},
```

Comment:
I’ve seen this a few places now. Does it just not work as intended at all?  Is it only escaping two slashes or?  Seems like it wouldn’t match super short strings either?

Reply:
Yeah, it just doesn't work. It's looking for a `'` not preceded by `\`, but `'\\'` is perfectly legitimate and fails the regex.

Reply:
\Yeah I get confused by all the extra escaping with '' strings.  Looks like exactly what you say.